cpp/include/cudf/stream_compaction.hpp

```diff
@@ -461,6 +463,7 @@ std::vector<std::unique_ptr<column>> filter(
   bool is_ptx,
   std::optional<void*> user_data             = std::nullopt,
   std::optional<std::vector<bool>> copy_mask = std::nullopt,
+  null_aware is_null_aware                   = null_aware::NO,
```

Comment:
Just to confirm my understanding: when it's null-aware, we handle output nulls based on the input nulls; otherwise, we throw errors. Is that correct?

Reply:
For my education

> we use a bitmask-and of the null-masks to determine the nullness of the output. We don't allocate a bitmask in the case that none of the input columns are actually nullable.

I thought this was the behavior when null-aware is enabled. Could you clarify what actually happens when it *is* null-aware?

Reply:
when it is null-aware the UDF is made to handle the null-ness of the inputs,
i.e:

a non-null-aware udf would be:

```cpp
void udf(int * output, int in0, int in1);
```

the null-aware equivalent would be:

```cpp
void udf(optional<int> * output, optional<int> in0, optional<int> in1)
```